### PR TITLE
Allow Drawer width to be customized

### LINKED ISSUE #99776: [Material] Allow Drawer width to be customized
## Use case

Being able to customize the width of the Material drawer widget. We're building a responsive app and find that we need to increase/decrease drawer width based on screen size.

## Proposal

`Drawer` already has a `width` field internally, it's just always set to a constant value (`_kWidth`). The proposed change is just adding the `width` field to the constructor and having it default to `_kWidth`, so this would be a backwards compatible change.

### PR DESCRIPTION
**Pull request by TytaniumDev**
Elevate the internal width parameter for the `Drawer` widget to the constructor, and default it to the previous constant value. More info on #99776 and #19724

Fixes #99776  and #19724

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by guidezpl — CHANGES_REQUESTED** [review]
The way we typically resolve this kind of value is 1. individual instance, 2. [`DrawerThemeData`](https://github.com/flutter/flutter/blob/7e9793dee1/packages/flutter/lib/src/material/drawer_theme.dart#L31), then 3. default. 

Could you model this change after the other properties, e.g. `backgroundColor`?

**Comment by TytaniumDev** [comment]
> The way we typically resolve this kind of value is 1. individual instance, 2. `DrawerThemeData`, then 3. default.
> 
> Could you model this change after the other properties, e.g. `backgroundColor`?

Can do! Is it okay if I add the existing `_kWidth` default to `DrawerThemeData`?

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

Great!

File: packages/flutter/test/material/drawer_test.dart [hunk 1]
```diff
@@ -445,4 +445,46 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('endDrawer'), findsNothing);
   });
+
+  testWidgets('Drawer width defaults to Material spec',
+      (WidgetTester tester) async {
```
Comment (darrenaustin) [inline packages/flutter/test/material/drawer_test.dart:450]: Minor nit, can these be on one line like the other tests?

**Review by darrenaustin — APPROVED** [review]
Sorry it took me so long to review this. Just had a minor nit. Nice work!

flutter-lgtm

**Pull request merged**